pkg/login/social/google_oauth_test.go

```diff
@@ -0,0 +1,492 @@
+package social
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/go-jose/go-jose/v3"
+	"github.com/go-jose/go-jose/v3/jwt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+func TestSocialGoogle_retrieveGroups(t *testing.T) {
+	type fields struct {
+		Scopes []string
+	}
+	type args struct {
+		client   *http.Client
+		userData *googleUserData
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []string
+		wantErr bool
+	}{
+		{
+			name: "No scope",
+			fields: fields{
+				Scopes: []string{},
+			},
+			args: args{
+				client: &http.Client{},
+				userData: &googleUserData{
+					Email: "test@example.com",
+				},
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "No groups",
```

Comment:
This test is currently the same as above - should add a scope and return empty list of groups.

Reply:
my bad, must have wrapped up for the day and forgot it